stock_batch_picking/report/batch_report.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+# © 2012-2014 Nicolas Bessi, Joel Grand-Guillaume, Alexandre Fayolle
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+import time
+from os.path import commonprefix
+from openerp.report import report_sxw
+from openerp import pooler, models
+
+_logger = logging.getLogger(__name__)
+
+
+class BatchAgregation(object):
+    """ Group operations from a single batch by source and dest locations
+    """
+
+    def __init__(self, batch_id, operations_by_loc):
+        self.batch_id = batch_id
+        self.operations_by_loc = operations_by_loc
+
+    def exists(self):
+        return False
+
+    @property
+    def picker_id(self):
+        return self.batch_id.picker_id
+
+    @property
+    def batch_name(self):
+        return self.batch_id.name
+
+    @property
+    def batch_notes(self):
+        return self.batch_id.notes or u''
+
+    def __hash__(self):
+        return hash(self.batch_id.id)
+
+    def __eq__(self, other):
+        return self.batch_id.id == other.batch_id.id
+
+    def iter_locations(self):
+        for locations in self.operations_by_loc:
+            offset = commonprefix(locations).rfind('/') + 1
+            display_locations = tuple(
+                loc[offset:].strip() for loc in locations)
+            yield display_locations, self._product_quantity(locations)
+
+    def _product_quantity(self, locations):
+        """iterate over the different products concerned by the operations for the
+        specified locations with their total quantity, sorted by product
+        default_code
+
+        locations: a tuple (source_location, dest_location)
+        """
+        products = {}
+        product_qty = {}
+        carrier = {}
+        operations = self.operations_by_loc[locations]
+        for operation in operations:
+            p_code = operation.product_id.default_code
+            products[p_code] = operation.product_id
+            carrier[p_code] = (
+                operation.picking_id.carrier_id and
+                operation.picking_id.carrier_id.partner_id.name or ''
+            )
+            if p_code not in product_qty:
+                product_qty[p_code] = operation.product_qty
+            else:
+                product_qty[p_code] += operation.product_qty
+        for p_code in sorted(products):
+            yield products[p_code], product_qty[p_code], carrier[p_code]
+
+
+class PrintBatch(report_sxw.rml_parse):
```

Comment:
Can you please split this into its own file? Makes things much more manageable from a high level, and is noted in contributing doc - https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming